apache/metron
# METRON-2060 Improving Alerts table config pane

## Contributor Comments

As a part of this PR I'm separating list visible in available columns in alerts table config pane. This makes a better user experience also fixes a few glitches.

List of changes:
- separating lists of visible and available columns 
- changing the width of the pane to make more space on the screen useful
- the layout of the pane automatically aligns to the width of the pane and the amount of content
- moving the lists beside each other (except in small displays)
- swapping the checkbox with add/remove buttons
- removing short name field from available columns
- applying the filter to the list of available columns

![Screen Shot 2019-04-05 at 11 54 42 AM](https://user-images.githubusercontent.com/2437400/55619829-ccda9280-5799-11e9-9903-e5060608fdc6.png)

![Screen Shot 2019-04-05 at 11 55 17 AM](https://user-images.githubusercontent.com/2437400/55619847-d401a080-5799-11e9-8f2b-acaf0b75634e.png)

![Screen Shot 2019-04-05 at 1 08 23 PM](https://user-images.githubusercontent.com/2437400/55623652-fef0f200-57a3-11e9-81d8-204b925146b5.png)

## Testing

- checkout this pr
- navigate to {PROJECT_ROOT}/metron-deployment/development/centos6
- start full dev wm (more in the Metron documentation)
- navigate to {PROJECT_ROOT}/metron-interface/metron-alerts/
- run ./scripts/start-dev.sh
- make sure the functionality described abowe works fine

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:


### For code changes:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```


### For documentation related changes:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.
